orderlyweb_api/orderly_web_api.py

```diff
@@ -16,7 +16,7 @@ def __init__(self, base_url, auth_provider_token):
 
     def run_report(self, report, params, timeout=600):
         url = 'reports/{}/run/?timeout={}'.format(report, timeout)
-        result = self.post(url, str(params))
+        result = self.post(url, str({'params': params}))
```

Comment:
Wary of pointing this out because it clearly works (given the test below) but looks a bit dicey. Seems to be relying on Python's string serialisation of dict - which happens to produce something that looks a bit like JSON but isn't really. For example, `str({'a', None})` produces `{'a': None}` which is not valid JSON: wrong quotes (and contains literal `None`).

This would be better:
```suggestion
        result = self.post(url, json.dumps({'params': params}))
```
Or, if requests is new enough then:
```
result = self.post(url, {'params': params})
```
and change `post()` to:
```
response = requests.post(url, json=data, headers=headers)
```
But this would be riskier, given that `publish_report` already uses `post()`.

Reply:
Good point, we've been relying on a well behaved user up til now (us!)